common/lib/util/defaults.js

```diff
@@ -116,16 +127,22 @@ Defaults.normaliseOptions = function(options) {
 	if(!('queueMessages' in options))
 		options.queueMessages = true;
 
-	var production = false;
 	if(options.restHost) {
 		options.realtimeHost = options.realtimeHost || options.restHost;
+		options.fallbackHosts = options.fallbackHostsUseDefault && !options.port && !options.tlsPort ? Defaults.FALLBACK_HOSTS : options.fallbackHosts;
 	} else {
 		var environment = (options.environment && String(options.environment).toLowerCase()) || Defaults.ENVIRONMENT;
-		production = !environment || (environment === 'production');
+		var production = !environment || (environment === 'production');
 		options.restHost = production ? Defaults.REST_HOST : environment + '-' + Defaults.REST_HOST;
 		options.realtimeHost = production ? Defaults.REALTIME_HOST : environment + '-' + Defaults.REALTIME_HOST;
```

Comment:
options.realtimeHost is mutated without checking if it contains a custom value.

Reply:
RTN17a AND RTN17b

Reply:
Yes that is a bug considering the current wording of the spec

Reply:
@lmars do you feel this requires a change to the spec, or to this library, please? From the discourse on this point so far (at least here) it feels like it could go either way. 😕

Reply:
It's a bug in the code here, should be gated with `if(!options.realtimeHost)`

Reply:
Are we assuming options.realtimeHost will be set?

Reply:
@sacOO7 I'm not sure I understand your question, `options.realtimeHost` is an optional value that can be specified in development environments where the realtime host cannot be otherwise inferred by the library, and if it is set then it should not be overridden as is done here, it should be gated with an if statement as mentioned previously.

Reply:
The identified bug has been addressed in https://github.com/ably/ably-js/pull/689.